crates/turborepo-lib/src/commands/login.rs

```diff
@@ -111,16 +121,25 @@ fn make_token_name() -> Result<String> {
 
 pub async fn login(base: &mut CommandBase) -> Result<()> {
     let repo_config = base.repo_config()?;
-    let login_url_base = repo_config.login_url();
-    debug!("turbo v{}", get_version());
-    debug!("api url: {}", repo_config.api_url());
-    debug!("login url: {login_url_base}");
-
     let redirect_url = format!("http://{DEFAULT_HOST_NAME}:{DEFAULT_PORT}");
-    let login_url = format!("{login_url_base}/turborepo/token?redirect_uri={redirect_url}");
+    let login_url_configuration = repo_config.login_url();
+    let mut login_url = Url::parse(login_url_configuration)?;
+
+    login_url
+        .path_segments_mut()
+        .map_err(|_: ()| Error::LoginUrlCannotBeABase {
+            value: login_url_configuration.to_string(),
+        })?
+        .extend(["turborepo", "token"]);
+
+    login_url
+        .query_pairs_mut()
+        .append_pair("redirect_uri", &redirect_url);
+
     println!(">>> Opening browser to {login_url}");
-    direct_user_to_url(&login_url);
     let spinner = start_spinner("Waiting for your authorization...");
+    direct_user_to_url(login_url.as_str());
```

Comment:
(Aligns `login` with `login_sso`, prepping these to be side-by-side diff-comparable before collapsing.)